cmd/contour/gatewayprovisioner.go

```diff
@@ -52,6 +52,10 @@ func registerGatewayProvisioner(app *kingpin.Application) (*kingpin.CmdClause, *
 		Default(config.metricsBindAddress).
 		StringVar(&config.metricsBindAddress)
 
+	cmd.Flag("leader-election-namespace", "The namespace in which the leader election resource will be created.").
+		Default(config.leaderElectionNamespace).
```

Comment:
hm, right now the default is empty since we just added this field in this PR and havent set a default here: https://github.com/projectcontour/contour/blob/a20cac5011b5c2a3614b542dd988a6fb5369d625/cmd/contour/gatewayprovisioner.go#L34-L41

since the example YAML configuration etc. sets up the provisioner to run in the `projectcontour` namespace by default we can probably set that as the default, (similar to the contour serve controller here https://github.com/projectcontour/contour/blob/a20cac5011b5c2a3614b542dd988a6fb5369d625/cmd/contour/serve.go#L130)

we should also note that the RBAC configuration needs to be updated if this flag is used in-cluster

Reply:
ok, i have not set the default value to `projectcontour`, is just want to keep it as before, I will update it

Reply:
> we should also note that the RBAC configuration needs to be updated if this flag is used in-cluster

I can't understand this, pls give me more info about it

Reply:
> > we should also note that the RBAC configuration needs to be updated if this flag is used in-cluster
> 
> I can't understand this, pls give me more info about it

See https://github.com/projectcontour/contour/blob/491e263e2ab3f4b6e6c0fa629b580491a03ab2da/examples/gateway-provisioner/01-roles.yaml#L179

When this flag is added, users should be aware this resource will have to change if they are using this flag in cluster etc., not a big thing but maybe nice to find a place to note it